src/ui/application-window.vala

```diff
@@ -369,6 +389,7 @@ namespace Peek.Ui {
             hide_handler = delay_indicator.hide.connect (() => {
               delay_indicator.disconnect (hide_handler);
               start_recording ();
+              update_time ();
```

Comment:
I would suggest to call this from inside `start_recording ()`. Also note that  `start_recording ()` gets called on two locations, currently your branch works only if one has configured displaying the start recording timeout.